components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/internal/AppManagerComponent.java

```diff
@@ -133,6 +135,26 @@ protected void activate(ComponentContext componentContext) throws Exception {
             int tenantId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
             AppManagerUtil.loadTenantWorkFlowExtensions(tenantId);
 
+            //register JWT implementation class as a OSGi service
+            String tokenGeneratorImplClazz = configuration.getFirstProperty(
+                    AppMConstants.TOKEN_GENERATOR_IMPL);
+            if (tokenGeneratorImplClazz == null) {
+                bundleContext.registerService(TokenGenerator.class.getName(), new JWTGenerator(),
+                                              null);
+            } else {
+                try {
```

Comment:
1. This needs to be done with proper OSGI service loading with scr.ref. Supply the internal Token Generator.
2. Move JWT logic into gateway compinent